### ISSUE TITLE
Nil value breaks following capture variables 

### ISSUE THREAD

**Issue by tonibony** [daddel80/notepadpp-multireplace#52]
This issue happens when searching with RE and replacing in "Use variables" mode. When a match is found, if one of the capture variables becomes nil, all subsequent capture variables also become nil, even if their corresponding subexpressions have matched.

Find what: `(this) *(is)? *(a match)`
Replace with: `set("CAP1 = <" .. (CAP1 or "?") .. ">, CAP2 = <" .. (CAP2 or "?") .. ">, CAP3 = <" .. (CAP3 or "?") .. ">")`
Use Variables: YES
Search Mode: Regular expression

Sample text:
```
this is a match
but is this a match
```

And here is the result:
```
CAP1 = <this>, CAP2 = <is>, CAP3 = <a match>
but is CAP1 = <this>, CAP2 = <?>, CAP3 = <?>
```

**Comment by daddel80** [comment]
Hi Tonibony,

Thanks for reporting the issue. I have already located the part in the code and will provide you with an update soon.

**Comment by tonibony** [comment]
Take your time, I won't need a fix anytime soon.

**Comment by daddel80** [comment]
Hi tonibony,

Thanks for catching that bug, even if you don't need the fix asap. Appreciate your help! Here’s the new [BUILD 472](https://github.com/daddel80/notepadpp-multireplace/actions/runs/10633913268). 
If you want, you can test the correction and let me know or simply close this issue.

I'll include this fix in the next Notepad++ release.

**Comment by daddel80** [comment]
fix will be available in MultiReplace Release 3.1.2.18
